drivers/include/sha5461as.h

```diff
@@ -0,0 +1,158 @@
+    SHA5461AS_ERR_DIG1_GPIO,    /**< Something went wrong with DIG1 GPIO */
+    SHA5461AS_ERR_DIG2_GPIO,    /**< Something went wrong with DIG2 GPIO */
+    SHA5461AS_ERR_DIG3_GPIO,    /**< Something went wrong with DIG3 GPIO */
+    SHA5461AS_ERR_DIG4_GPIO,    /**< Something went wrong with DIG4 GPIO */
+    SHA5461AS_ERR_DIGITS,       /**< Something went wrong with digits value */
+};
+
+/**
+ * @brief   Device initialization parameters
+ */
+typedef struct {
+    gpio_t a;       /**< Data input pin */
+    gpio_t b;       /**< Data input pin */
+    gpio_t c;       /**< Data input pin */
+    gpio_t d;       /**< Data input pin */
+    gpio_t e;       /**< Data input pin */
+    gpio_t f;       /**< Data input pin */
+    gpio_t g;       /**< Data input pin */
+    gpio_t dp;      /**< Data input pin */
+    gpio_t dig1;    /**< Data input pin */
+    gpio_t dig2;    /**< Data input pin */
+    gpio_t dig3;    /**< Data input pin */
+    gpio_t dig4;    /**< Data input pin */
+} sha5461as_params_t;
+
+/**
+ * @brief   Device descriptor for the SHA5461AS
+ */
+typedef struct {
+    sha5461as_params_t params;  /**< Device parameters */
+    uint32_t value;             /**< Binary value to display */
+    uint8_t digits;             /**< Number of digits */
+    uint8_t current_digit;      /**< Displayed digit */
+    int status;                 /**< Status of the timer_periodic for this dev */
+} sha5461as_t;
+
+/**
+ * @brief   Initialize the given DSP0401
+ *
+ * @param[out] dev      Initialized device descriptor of SHA5461AS device
+ * @param[in]  params   Device parameters to use
+ *
+ * @return              SHA5461AS_OK if everything is good
+ * @return              -SHA5461AS_ERR_A_GPIO if an error occurred during A GPIO
+ * @return              -SHA5461AS_ERR_B_GPIO if an error occurred during B GPIO
+ * @return              -SHA5461AS_ERR_C_GPIO if an error occurred during C GPIO
+ * @return              -SHA5461AS_ERR_D_GPIO if an error occurred during D GPIO
+ * @return              -SHA5461AS_ERR_E_GPIO if an error occurred during E GPIO
+ * @return              -SHA5461AS_ERR_F_GPIO if an error occurred during F GPIO
+ * @return              -SHA5461AS_ERR_G_GPIO if an error occurred during G GPIO
+ * @return              -SHA5461AS_ERR_DP_GPIO if an error occurred during DP GPIO
+ * @return              -SHA5461AS_ERR_DIG1_GPIO if an error occurred during DIG1 GPIO
+ * @return              -SHA5461AS_ERR_DIG2_GPIO if an error occurred during DIG2 GPIO
+ * @return              -SHA5461AS_ERR_DIG3_GPIO if an error occurred during DIG3 GPIO
+ * @return              -SHA5461AS_ERR_DIG4_GPIO if an error occurred during DIG4 GPIO
+ */
+int sha5461as_init(sha5461as_t *dev, const sha5461as_params_t *params);
+
+/**
+ * @brief   Shift the current digit off and the next one on
+ *
+ * @param[in] dev       Initialized device descriptor of SHA5461AS device
+ *
+ */
+int sha5461as_shift(sha5461as_t *dev);
```

Comment:
It can be, but I think the user has the right to use digit7seg_poweron with the timer or handle it themselves with a loop.

Reply:
I would prefer that this not be part of the driver's public API (and therefore made static). The only reason I can think of to expose this would be if you were doing super-loop style programming. A smaller API will make the driver more maintainable.